specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2022-08-01/computeRPCommon.json

```diff
@@ -1053,6 +1053,18 @@
             "lun"
           ],
           "description": "Specifies the parameters that are used to add a data disk to a virtual machine. <br><br> For more information about disks, see [About disks and VHDs for Azure virtual machines](https://docs.microsoft.com/azure/virtual-machines/managed-disks-overview)."
+        },
+        "diskControllerType": {
+          "type": "string",
+          "description": "Specifies the disk controller type configured for the VM. This property is only supported for virtual machines whose operating system disk and and VM size supports hyperVGeneration: V2. <br> For more information about Disk Controller Types supported please refer to https://aka.ms/azure-diskcontrollertypes.<br><br>NOTE: This property will be set to the default disk controller type if not specified provided virtual machine is being created as a hyperVGeneration: V2 based on the capabilities of the operating system disk and VM size from the the specified minimum api version. <br>You need to deallocate the VM before updating its disk controller type unless you are updating the VM size in the VM configuration which implicitly deallocates and reallocates the VM. <br><br> Minimum api-version: 2022-08-01",
```

Comment:
Need to update the following statement as "This property is only supported for virtual machines whose operating system disk and VM sku supports Generation 2 (https://docs.microsoft.com/en-us/azure/virtual-machines/generation-2), pls. check the values returned for VM sku capabilities HyperVGenerations in the response of Microsoft.Compute SKUs api in a region (https://docs.microsoft.com/rest/api/compute/resourceskus/list) contains V2.

Reply:
Will update